### PR TITLE
Fixed the behavior around API quota limits

### PR DESCRIPTION
**Pull request by guykeller**
## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: https://github.com/demisto/etc/issues/23385

## Description
A bad error message would be thrown when the API quota was reached if the "Disregard quota errors" was set to True.
This is now fixed.

## Does it break backward compatibility?
   - [ ] No

## Must have
- [ ] Tests
- [ ] Documentation 

## Demisto Partner?
- [ ] The title must be in the following format: **[YOUR_PARTNER_ID] short description**

### CODE REVIEW

**Comment by guykeller** [comment]
Build with unit test failing on the same exact error that was described in the issue:
https://circleci.com/gh/demisto/content/48212#config/containers/1

File: Integrations/AbuseDB/AbuseDB_test.py [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+    __setattr__ = dict.__setitem__
+    __delattr__ = dict.__delitem__
+
+
+api_quota_reached_request_response = {
+    'status_code': 429
+}
+api_quota_reached_request_response_with_dot_access = dotdict(api_quota_reached_request_response)
+
+API_QUOTA_REACHED_TEST_CASES = (
```
Comment (ronykoz) [inline Integrations/AbuseDB/AbuseDB_test.py:20]: ?

File: Integrations/AbuseDB/AbuseDB.py [hunk 2]
```diff
@@ -10,6 +10,7 @@
 
 ''' GLOBALS '''
 VERBOSE = True
+# raise ValueError(demisto.params())
```
Comment (ronykoz) [inline Integrations/AbuseDB/AbuseDB.py:13]: remove

Comment (ronykoz) [inline Integrations/AbuseDB/AbuseDB.py:13]:
```suggestion
```

**Comment by ronykoz** [comment]
Fix conflict's and merge

**Comment by guykeller** [comment]
Closing this  and opening a new PR.

**Pull request closed without merging**